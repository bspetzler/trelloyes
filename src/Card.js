import React from 'react';

function Card(props){
    return (
        <div className='Card'>
            <button onClick={() => props.onDelete(props.id)} type="button">delete</button>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
        </div>
    )
}

Card.propTypes = {
    onClickDelete: () => {}
  }

export default Card;