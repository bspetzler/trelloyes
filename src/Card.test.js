import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card.js';

describe ('messages-component', () => {
    it('render without crashing', () =>{
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('render the UI as expected', ()=>{
        const tree = renderer
        .create(<Card title='First card' content='lorem ipsum'/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('render the UI as expected', ()=>{
        const tree = renderer
        .create(<Card title='Second card' content='lorem ipsum'/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})
