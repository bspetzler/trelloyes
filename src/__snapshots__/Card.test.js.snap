// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages-component render the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;

exports[`messages-component render the UI as expected 2`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Second card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;
